homeassistant/components/unifiprotect/entity.py

```diff
@@ -203,3 +220,99 @@ def _async_update_device_from_protect(self) -> None:
             self.device = self.data.api.bootstrap.nvr
 
         self._attr_available = self.data.last_update_success
+
+
+class AccessTokenMixin(Entity):
+    """Adds access_token attribute and provides access tokens for use for anonymous views."""
+
+    @property
+    def access_tokens(self) -> deque[str]:
+        """Get valid access_tokens for current entity."""
+        assert isinstance(self, ProtectDeviceEntity)
```

Comment:
Why don't we inherit `ProtectDeviceEntity`?

Reply:
It is to prevent a multiple inheritance issue. `EventThumbnailMixin` and `AccessTokenMixin` are both inheriting an ABC so none of the `ProtectDeviceEntity` functionality gets duplicated in them.

Reply:
In `ProtectEventSensor` I actually explicitly call a method from this class so to override a `ProtectDeviceEntity` as well.

Reply:
Scratch that, that is in the refactor PR. I can see what I can do to make this class inherit `ProtectDeviceEntity` as well.

Reply:
Adjusted in the ufp_value PR since I already was changing around AccessToken/EventThumbnail classes.

Will get the other 2 in a separate follow up PR though.